build-tools/packages/build-cli/src/library/package.ts

```diff
@@ -754,10 +754,10 @@ async function findDepUpdates(
 
 		// If we're allowing pre-release, use the next tagged version. Warn if it is lower than the latest.
 		if (prerelease) {
-			dependencyVersionMap[pkgName] = next;
-			if (semver.gt(latest, next)) {
+			dependencyVersionMap[pkgName] = dev;
```

Comment:
I think the comment above this needs an update too. I'd make it more explicit too, for people not familiar with this part of the codebase (like me before looking at this PR). Maybe `If we're allowing pre-release, use the version that has the 'dev' dist-tag in npm. Warn if it is lower than the latest.`